Please move -Werror to your CI

Please revert 2d8d8902aff5e8b8ca97113a8b3d8e0084d20237 (#134). This project is widely packaged in Linux distros. Some of those distros aggressively adopt new compiler versions and therefore will see build warnings that aren't yet seen in your environment.

Instead, I recommend you use `Werror` in your CI so that it helps you ensure your desired level of code quality without causing difficulty building this project in other environments.

For instance, I got this error in Debian Unstable which is using gcc-15:

```c
src/lib/ini.c: In function ‘ini_parse_file’:
src/lib/ini.c:93:5: error: ‘__builtin_strncpy’ specified bound 4097 equals
 destination size [-Werror=stringop-truncation]
   93 |     strncpy(dest, src, size);
      |     ^
cc1: all warnings being treated as errors
````

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,9 +71,15 @@ endif()
 
 # Use high warning levels
 if(MSVC)
-  add_compile_options(/W4 /WX)
+    add_compile_options(/W4)
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        add_compile_options(/WX)
+    endif()
 else()
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+    add_compile_options(-Wall -Wextra -Wpedantic)
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        add_compile_options(-Werror)
+    endif()
 endif()
 
 add_subdirectory(lib)

